needle.pm

```diff
@@ -51,20 +52,32 @@ sub new {
     }
 
     my $self = {};
-    if (index($jsonfile, $bmwqemu::vars{PRJDIR}) == 0) {
-        $self->{file} = substr($jsonfile, length($bmwqemu::vars{PRJDIR}) + 1);
-    }
-    elsif (-f File::Spec->catfile($bmwqemu::vars{PRJDIR}, $jsonfile)) {
-        # json file path already relative
-        $self->{file} = $jsonfile;
-        $jsonfile = File::Spec->catfile($bmwqemu::vars{PRJDIR}, $jsonfile);
-    }
-    else {
-        die "Needle $jsonfile is not under project directory $bmwqemu::vars{PRJDIR}";
-    }
 
-    # $json->{file} contains path relative to $bmwqemu::vars{PRJDIR}
-    # $jsonfile contains absolute path within $bmwqemu::vars{PRJDIR}
+    # locate the needle file in one of the lookup directories
+    # - This code initializes $json->{file} so it contains the path relative to the lookup directory and re-assigns
+    #   $jsonfile to contain the absolute path.
+    # - A custom needle repo checkout referenced by the $bmwqemu::vars{NEEDLES_DIR} variable is considered to support loading
+    #   custom needles with the openqa-clone-custom-git-refspec script. The custom needle repo must be within the current working
```

Comment:
Better stay with "os-autoinst speak" only, i.e. just reference NEEDLES_DIR, don't care about openQA tools, or just mention them as example, e.g. as you did one line below